osu.Game.Tests/Visual/Navigation/TestSceneInterProcessCommunication.cs

```diff
@@ -0,0 +1,86 @@
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Overlays;
+
+namespace osu.Game.Tests.Visual.Navigation
+{
+    [TestFixture]
+    [Ignore("This test cannot be run headless, as it requires the game host running the nested game to have IPC bound.")]
+    public class TestSceneInterProcessCommunication : OsuGameTestScene
+    {
+        private HeadlessGameHost ipcSenderHost = null!;
+
+        private OsuSchemeLinkIPCChannel osuSchemeLinkIPCReceiver = null!;
+        private OsuSchemeLinkIPCChannel osuSchemeLinkIPCSender = null!;
+
+        private const int requested_beatmap_set_id = 1;
+
+        [Resolved]
+        private GameHost gameHost { get; set; } = null!;
+
+        public override void SetUpSteps()
+        {
+            base.SetUpSteps();
+            AddStep("set up request handling", () =>
+            {
+                ((DummyAPIAccess)API).HandleRequest = request =>
+                {
+                    switch (request)
+                    {
+                        case GetBeatmapSetRequest gbr:
+
+                            var apiBeatmapSet = CreateAPIBeatmapSet();
+                            apiBeatmapSet.OnlineID = requested_beatmap_set_id;
+                            apiBeatmapSet.Beatmaps = apiBeatmapSet.Beatmaps.Append(new APIBeatmap
+                            {
+                                DifficultyName = "Target difficulty",
+                                OnlineID = 75,
+                            }).ToArray();
+                            gbr.TriggerSuccess(apiBeatmapSet);
+                            return true;
+                    }
+
+                    return false;
+                };
+            });
+            AddStep("create IPC receiver channel", () => osuSchemeLinkIPCReceiver = new OsuSchemeLinkIPCChannel(gameHost, Game));
+            AddStep("create IPC sender channel", () =>
+            {
+                ipcSenderHost = new HeadlessGameHost(gameHost.Name, new HostOptions { BindIPC = true });
+                osuSchemeLinkIPCSender = new OsuSchemeLinkIPCChannel(ipcSenderHost);
+            });
+        }
+
+        [Test]
+        public void TestOsuSchemeLinkIPCChannel()
+        {
+            AddStep("open beatmap via IPC", () => osuSchemeLinkIPCSender.HandleLinkAsync($@"osu://s/{requested_beatmap_set_id}").WaitSafely());
+            AddUntilStep("beatmap overlay displayed", () => Game.ChildrenOfType<BeatmapSetOverlay>().FirstOrDefault()?.State.Value == Visibility.Visible);
+            AddUntilStep("beatmap overlay showing content", () => Game.ChildrenOfType<BeatmapSetOverlay>().FirstOrDefault()?.Header.BeatmapSet.Value.OnlineID == requested_beatmap_set_id);
+        }
+
+        public override void TearDownSteps()
+        {
+            AddStep("dispose IPC receiver", () => osuSchemeLinkIPCReceiver.Dispose());
+            AddStep("dispose IPC sender", () =>
+            {
+                osuSchemeLinkIPCReceiver.Dispose();
```

Comment:
```suggestion
                osuSchemeLinkIPCSender.Dispose();
```

Reply:
Can probably combine both into a single step to be honest.